### ISSUE TITLE
Add Shellcheck Linter

### ISSUE THREAD

**Issue by danehans** [envoyproxy/gateway#314]
Scripts continue to be added to the project, mainly to support Make targets. Shellcheck should be used to for static analysis and lint tool, for these scripts.

**Comment by danehans** [comment]
Reopening since I see the following error when running `make lint` locally:
```
...
Running Shellcheck linter ...
/bin/sh: shellcheck: command not found
make[1]: *** [lint.shellcheck] Error 127
```

I understand that spellchecker is installed in GitHub Actions by default but Devs should be able to run the target locally without error.

**Comment by Xunzhuo** [comment]
Right, shelchecker is pre-installed in GA, we can add it in local.

**Comment by danehans** [comment]
> ... we can add it in local.

@Xunzhuo are you able to submit a PR to add local support of `spellcheck`?

**Comment by Xunzhuo** [comment]
Ok, sure. @danehans
